CrowdFundingApplication/Controllers/ProfileController.cs

```diff
@@ -0,0 +1,94 @@
+﻿namespace CrowdFundingApplication.Controllers;
+
+public class ProfileController : Controller
+{
+    private readonly UserManager<User> _userManager;
+    private readonly IWebHostEnvironment _webHostEnvironment;
+
+    public ProfileController(UserManager<User> userManager, IWebHostEnvironment webHostEnvironment)
+    {
+        _userManager = userManager;
+        _webHostEnvironment = webHostEnvironment;
+    }
+
+    public async Task<IActionResult> Index()
+    {
+        var user = await _userManager.FindByNameAsync(HttpContext.User.Identity.Name);
+        return View(user);
+    }
+
+    [HttpGet]
+    public IActionResult Settings() => View();
+
+    [HttpPost]
+    public async Task<IActionResult> Settings(EditProfileViewModel model, IFormFile Image)
+    {
+        if (ModelState.IsValid)
+        {
+            var user = await _userManager.FindByNameAsync(HttpContext.User.Identity.Name);
+            if (user is null)
+            {
+                return View(model);
+            }
+
+            //update dates for user
+            if (model.Age is not null) user.Age = (int)model.Age;
+            if (model.Email is not null) user.Email = model.Email;
```

Comment:
You can't just change the email - look at this table in the database and you will understand what I mean.

Reply:
Has been edited